src/fcntl.rs

```diff
@@ -141,7 +141,7 @@ libc_bitflags!(
         #[cfg(any(
             freebsdlike,
             linux_android,
-            solarish,
+            target_os = "illumos",
```

Comment:
According to the [error log](https://github.com/nix-rust/nix/issues/2540#issue-2664987582), this `O_DIRECT` thing is not the only missing symbol, it also needs `SIGINFO`.

Are these symbols not available on the target?